Bugfixes for COOL-VENOM site-to-site tunnel

jsf9k opened the pull request:
## 🗣 Description ##

This pull request contains a few bug fixes that came up during the COOL/VENOM debugging window on March 3, 2021:
1. Use AES-256 (CBC) encryption vice AES-256-GCM-16
2. Create static routes for both the site-to-site VPN and default Transit Gateway route tables that direct the appropriate traffic to the VPN.

## 💭 Motivation and context ##

These changes were found to be required because:
1. The VPN hardware being used on the VENOM end does not support AES-256-GCM-16.
2. We cannot automatically propagate these routes because we do not use [BGP](https://en.wikipedia.org/wiki/Border_Gateway_Protocol).

## 🧪 Testing ##

1. The site-to-site VPN tunnel is able to connect with this change.
2. We are able to resolve against VENOM's DNS with these static routes.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
* [x] All new and existing tests pass.

jsf9k committed:
Use AES-256 encryption instead of AWS-256-GCM-16

AES-256 uses CBC, which is required for the VENOM tunnel.

jsf9k committed:
Swap the order of keys and values in the venom_cidrs variable

This is needed to retrieve individual CIDRs for static routes that
will need to be added to Transit Gateway route tables.

jsf9k committed:
Add a static route (pointing to the VPN) to the VPN's TGW route table

Because we don't use BGP we can't automatically propagate such a
route.

jsf9k committed:
Add a static route to the VPN to the default TGW route table

Since we don't use BGP, we cannot automatically propagate such a
route.

jsf9k committed:
Add ping reply ingress and egress rules for the VENOM security group

Also separate the port key into separate from_port and to_port keys.
This is necessary because, in the case of ICMP, the from port is
treated as the ICMP type and the to port is treated as the ICMP code.

mcdonnnj commented on README.md:
```diff
@@ -49,9 +49,9 @@ or
 | provisionvenom_policy_name | The name to assign the IAM policy that allows provisioning of the VENOM layer in the Shared Services account. | `string` | `ProvisionVenom` | no |
 | tags | Tags to apply to all AWS resources created. | `map(string)` | `{}` | no |
-| venom_cidrs | A map with keys equal to the VENOM CIDR blocks and values equal to a brief description (e.g. {"10.200.0.0/16": "Primary", "10.201.0.0/16": "Secondary"}). | `map(string)` | n/a | yes |
+| venom_cidrs | A map with values equal to the VENOM CIDR blocks and keys equal to a brief description (e.g. {"Primary": "10.200.0.0/16", "Secondary": "10.201.0.0/16"}). | `map(string)` | n/a | yes |
```
if we're going to use a hard-coded key per
https://github.com/cisagov/cool-sharedservices-venom/blob/f8591be1f40eb2457a9402859027c5fde58367ea/transitgateway.tf#L33
and
https://github.com/cisagov/cool-sharedservices-venom/blob/f8591be1f40eb2457a9402859027c5fde58367ea/transitgateway.tf#L43
, then I think we should use that in the description (or a call out in the README somewhere). There's no indication here that a specific key is required.

jsf9k replied:
That prefix list was a failed attempt to pre-emptively support VENOM-East and VENOM-West.  I can't really do that properly right now since VENOM-West doesn't exist yet, so I removed the prefix list in commit a0c4708 and am using a simple CIDR block instead.

When I re-ran `terraform apply` with the changes in commit a0c4708 I verified that it only:
* Removed a few permissions from the policy related to creating and managing prefix lists
* Removed the prefix list
* Modified a bunch of security group rules to use the CIDR instead of the prefix list

mcdonnnj replied:
Ahh, thanks for the background information.

mcdonnnj reviewed (approved):
Thanks for your continued work in getting this functional. ✔

dav3r reviewed (approved):
🐍 Looksssssss good to me!

jsf9k committed:
Remove prefix list for VENOM and just use a CIDR block instead

The use of the prefix list was an attempt to prepare for the future by standing up separate tunnels for VENOM-East and VENOM-West.  That jazz isn't ready to go yet, so I'll just go ahead and simplify things until then.

dav3r reviewed (approved):
Ssssstill good! 🐍 👍

jsf9k committed:
Remove ping rules from VENOM security group

I verified that ping is disabled on these instances as part of the hardening required for our ATO.

mcdonnnj reviewed (approved):
Even better the second time! 🌟
